### ISSUE TITLE
Index was out of range

### ISSUE THREAD

**Issue by danielrey204** [GralDispersionModel/GRAMM#22]
**Error message:**
```
Unhandled exception. System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at GRAMM_2001.Program.Main(String[] args) in D:\GitHub\GRAMM\src\Program.cs:line 581
```
Program.cs line 581: 
```
//store last time step
STOREcalculate(NX, NY, NZ);
```

**Steps that lead to this issue:**
I am running GRAMM for Windows under the GRAL GUI version V2311.
I defined the GRAL and GRAMM domains, added a source and its associated emissions, domain land use, topography and meterology files as requested. 

After creating the GRAMM grid and adding the topography file this is the domain cell distribution:
```
Minimum elevation: 22
Maximum elevation: 255
Number of cells in E-W direction: 65
Cell length in E-W direction: 120
E-W extents: 7800
Number of cells in S-N direction: 52
Cell length in S-N direction: 120
S-N extents: 6240
Z1= 22 m
Z2= 32 m
Z3= 46 m
Z4= 65,6 m
Z5= 93 m
Z6= 131,5 m
Z7= 185,2 m
Z8= 260,5 m
Z9= 365,9 m
Z10= 513,5 m
Z11= 720,1 m
Z12= 1009,4 m
Z13= 1414,3 m

```
Then, at the _computation_ tab i get the "ok" for all Control files, Meteo files, Emission files and Obstacle files. When I play the GRAMM model it stops giving the abovementioned issue. 

Do you have an idea of what this might be happening and what am I missing?

Thank you in advance,
Daniel

**Comment by MarkusKuntner** [comment]
I have also set up a model with a cell size of 150 m and have not found any problems. 
Is it possible that there is a problem with the Corine data? Or could it be that receptor points are located beyond the domain area? Is it possible to compile GRAMM as a debug version and then find the exact location of the bug?

**Comment by danielrey204** [comment]
Hi @MarkusKuntner , 
Thank you for your answer. I checked the land use data and found no issues. I still didn't add receptors to the area. So what I did was to erase the project and begin it again from scratch, and that did the trick. I have no idea what changed, but since when I began this project it was also my first time with GRAL/GRAMM, probably I generated some file at the beginning (wrong domain maybe?) that didn't got overwritten and somehow was causing the issue. 

Right now I manged to run GRAMM at 99.9% before the simulation stopped, but it arrived to the last flow situation so everything seems ok. 

Thanks again!
Dani
